OSDOCS#5371: Document the explicit list of required credential permissions for GCP (#56118)
openshift/openshift-docs: modules/installation-gcp-permissions.adoc

ifeval::["{context}" == "installing-gcp-user-infra-vpc"]
:template:
endif::[]
ifeval::["{context}" == "installing-gcp-account"]
:ipi:
endif::[]

> 

Author: Addressed

* Service Account User
* Storage Admin

> The following are the required permissions for installer-provisioned GCP infrastructure for creating and deleting the {product-title} cluster.

Reviewer: ```suggestion
The following are the required permissions for provisioning GCP infrastructure for creating and deleting the {product-title} cluster.
```

Author: Acked

====
* `storage.buckets.create`
* `storage.buckets.get`
* `storage.buckets.delete`
* `storage.buckets.list`
* `storage.objects.create`
* `storage.objects.delete`
* `storage.objects.get`
* `storage.objects.list`
====

.Required permissions for creating health check resources
[%collapsible]
====
* `compute.healthChecks.create`
* `compute.healthChecks.get`
* `compute.healthChecks.list`
* `compute.healthChecks.useReadOnly`
* `compute.httpHealthChecks.create`
* `compute.httpHealthChecks.get`
* `compute.httpHealthChecks.list`
* `compute.httpHealthChecks.useReadOnly`
====

.Required permissions to get GCP zone and region related information
[%collapsible]
====
* `compute.globalOperations.get`
* `compute.regions.list`
* `compute.regionOperations.get`
* `compute.zones.get`
* `compute.zones.list`
* `compute.zoneOperations.get`
====

.Additional permissions for checking services and quotas
[%collapsible]
====
* `monitoring.timeSeries.list`
* `serviceusage.quotas.get`
* `serviceusage.services.list`
====

.Additional IAM permissions for installation
[%collapsible]
====
* `iam.roles.get`
====

.Additional Images permissions for installation
[%collapsible]
====
* `compute.images.list`
====

> 

Reviewer: ```suggestion

.Additional permission for running gather bootstrap
[%collapsible]
====
* `compute.instances.getSerialPortOutput`
====
```

Author: Acked

.Required permissions for deleting network resources
[%collapsible]
====
* `compute.addresses.delete`
* `compute.addresses.deleteInternal`
* `compute.addresses.list`
* `compute.firewalls.delete`
* `compute.firewalls.list`
* `compute.forwardingRules.delete`
* `compute.forwardingRules.list`
* `compute.networks.delete`
* `compute.networks.list`
* `compute.networks.updatePolicy`
* `compute.routers.delete`
* `compute.routers.list`
* `compute.routes.list`
* `compute.subnetworks.delete`
* `compute.subnetworks.list`
====

.Required permissions for deleting load balancer resources
[%collapsible]
====
* `compute.regionBackendServices.delete`
* `compute.regionBackendServices.list`
* `compute.targetPools.delete`
* `compute.targetPools.list`
====

.Required permissions for deleting DNS resources
[%collapsible]
====
* `dns.changes.create`
* `dns.managedZones.delete`
* `dns.managedZones.get`
* `dns.managedZones.list`
* `dns.resourceRecordSets.delete`
* `dns.resourceRecordSets.list`

> ====

Reviewer: ```suggestion
ifdef::template[]
* `dns.resourceRecordSets.update`
endif::template[]
====
```

Author: Acked

.Required permissions for deleting Service Account resources
[%collapsible]
====
* `iam.serviceAccounts.delete`
* `iam.serviceAccounts.get`
* `iam.serviceAccounts.list`
* `resourcemanager.projects.getIamPolicy`
* `resourcemanager.projects.setIamPolicy`
====

.Required permissions for deleting compute resources
[%collapsible]
====
* `compute.disks.delete`
* `compute.disks.list`
* `compute.instanceGroups.delete`
* `compute.instanceGroups.list`
* `compute.instances.delete`
* `compute.instances.list`
* `compute.instances.stop`
* `compute.machineTypes.list`
====

.Required for deleting storage resources
[%collapsible]
====
* `storage.buckets.delete`
* `storage.buckets.getIamPolicy`
* `storage.buckets.list`
* `storage.objects.delete`
* `storage.objects.list`
====

.Required permissions for deleting health check resources
[%collapsible]
====
* `compute.healthChecks.delete`
* `compute.healthChecks.list`
* `compute.httpHealthChecks.delete`
* `compute.httpHealthChecks.list`
====

> .Additional Images permissions for installation

Reviewer: ```suggestion
.Additional Images permissions for deletion
```

Author: Acked

[%collapsible]
====
* `compute.images.list`
====

.Additional permissions for running bootstrap
[%collapsible]
====
* `compute.instances.getSerialPortOutput`
====

> 

Reviewer: ```suggestion

ifdef::template[]
.Required permissions for deleting image resources
[%collapsible]
====
* `compute.images.delete`
====
endif::template[]
```

Author: Acked

> .Additional permissions for running bootstrap
> [%collapsible]
> ====
> * `compute.instances.getSerialPortOutput`
> ====
> 

Reviewer: ```suggestion

```

Author: Acked

* Service Account User
* Storage Admin

The following are the required permissions for provisioning GCP infrastructure for creating and deleting the {product-title} cluster.

.Required permissions for creating network resources
[%collapsible]
====
* `compute.addresses.create`
* `compute.addresses.createInternal`
* `compute.addresses.get`
* `compute.addresses.delete`
* `compute.addresses.list`
* `compute.addresses.use`
* `compute.addresses.useInternal`
* `compute.firewalls.create`
* `compute.firewalls.get`
* `compute.firewalls.delete`

> * `compute.firewalls.list`

Reviewer 2: Suggest to rearrange the permissions in ascending order as below
* `compute.firewalls.create`
* `compute.firewalls.delete`
* `compute.firewalls.get`
* `compute.firewalls.list`

Author: Addressed

* `compute.forwardingRules.create`
* `compute.forwardingRules.get`
* `compute.forwardingRules.list`
* `compute.forwardingRules.setLabels`
* `compute.networks.updatePolicy`
* `compute.networks.create`
* `compute.networks.get`

> * `compute.networks.list`

Author: Addressed

* `compute.routers.create`
* `compute.routers.get`
* `compute.routers.list`
* `compute.routers.update`
* `compute.routes.list`
* `compute.subnetworks.create`
* `compute.subnetworks.get`
* `compute.subnetworks.list`
* `compute.subnetworks.use`
* `compute.subnetworks.useExternalIp`
====

.Required permissions for creating load balancer resources
[%collapsible]
====
* `compute.regionBackendServices.create`
* `compute.regionBackendServices.get`
* `compute.regionBackendServices.list`
* `compute.regionBackendServices.update`
* `compute.regionBackendServices.use`
* `compute.targetPools.addInstance`
* `compute.targetPools.create`
* `compute.targetPools.get`
* `compute.targetPools.list`
* `compute.targetPools.removeInstance`
* `compute.targetPools.use`
====

.Required permissions for creating DNS resources
[%collapsible]
====
* `dns.changes.get`

> * `dns.changes.create`

Reviewer 2: Suggest to rearrange the permissions in ascending order as below
* `dns.changes.create`
* `dns.changes.get`

Author: Addressed

====
* `iam.serviceAccountKeys.create`
* `iam.serviceAccountKeys.delete`
* `iam.serviceAccountKeys.get`
* `iam.serviceAccountKeys.list`
* `iam.serviceAccounts.actAs`
* `iam.serviceAccounts.create`
* `iam.serviceAccounts.delete`
* `iam.serviceAccounts.get`
* `iam.serviceAccounts.list`
* `resourcemanager.projects.get`
* `resourcemanager.projects.getIamPolicy`
* `resourcemanager.projects.setIamPolicy`
====

.Required permissions for creating compute resources
[%collapsible]
====
* `compute.disks.create`
* `compute.disks.get`
* `compute.disks.list`
* `compute.instanceGroups.create`
* `compute.instanceGroups.delete`
* `compute.instanceGroups.get`
* `compute.instanceGroups.list`
* `compute.instanceGroups.update`
* `compute.instanceGroups.use`
* `compute.instances.create`
* `compute.instances.delete`
* `compute.instances.get`
* `compute.instances.list`
* `compute.instances.setLabels`
* `compute.instances.setMetadata`
* `compute.instances.setServiceAccount`
* `compute.instances.setTags`
* `compute.instances.use`
* `compute.machineTypes.get`
* `compute.machineTypes.list`
====

.Required for creating storage resources
[%collapsible]
====
* `storage.buckets.create`
* `storage.buckets.get`

> * `storage.buckets.delete`

Author: Addressed

* `storage.buckets.list`
* `storage.objects.create`
* `storage.objects.delete`
* `storage.objects.get`
* `storage.objects.list`
====

.Required permissions for creating health check resources
[%collapsible]
====
* `compute.healthChecks.create`
* `compute.healthChecks.get`
* `compute.healthChecks.list`
* `compute.healthChecks.useReadOnly`
* `compute.httpHealthChecks.create`
* `compute.httpHealthChecks.get`
* `compute.httpHealthChecks.list`
* `compute.httpHealthChecks.useReadOnly`
====

.Required permissions to get GCP zone and region related information
[%collapsible]
====
* `compute.globalOperations.get`
* `compute.regions.list`
* `compute.regionOperations.get`
* `compute.zones.get`
* `compute.zones.list`

> * `compute.zoneOperations.get`

Reviewer 2: Suggest to rearrange the permissions in ascending order as below
* `compute.globalOperations.get`
* `compute.regionOperations.get`
* `compute.regions.list`
* `compute.zoneOperations.get`
* `compute.zones.get`
* `compute.zones.list`

Author: Addressed

====

> 

Author: Addressed

[%collapsible]
====
* `compute.images.list`
====

.Optional permission for running gather bootstrap
[%collapsible]
====
* `compute.instances.getSerialPortOutput`
====

.Required permissions for deleting network resources
[%collapsible]
====
* `compute.addresses.delete`
* `compute.addresses.deleteInternal`
* `compute.addresses.list`
* `compute.firewalls.delete`
* `compute.firewalls.list`
* `compute.forwardingRules.delete`
* `compute.forwardingRules.list`
* `compute.networks.delete`
* `compute.networks.list`
* `compute.networks.updatePolicy`
* `compute.routers.delete`
* `compute.routers.list`
* `compute.routes.list`
* `compute.subnetworks.delete`
* `compute.subnetworks.list`
====

.Required permissions for deleting load balancer resources
[%collapsible]
====
* `compute.regionBackendServices.delete`
* `compute.regionBackendServices.list`
* `compute.targetPools.delete`
* `compute.targetPools.list`
====

.Required permissions for deleting DNS resources
[%collapsible]
====
ifdef::template[]
* `dns.resourceRecordSets.update`
endif::template[]
* `dns.changes.create`
* `dns.managedZones.delete`
* `dns.managedZones.get`
* `dns.managedZones.list`
* `dns.resourceRecordSets.delete`

> * `dns.resourceRecordSets.list`

Author: Addressed

* Service Account User
* Storage Admin

The following are the required permissions for provisioning GCP infrastructure for creating and deleting the {product-title} cluster.

.Required permissions for creating network resources
[%collapsible]
====
* `compute.addresses.create`
* `compute.addresses.createInternal`
* `compute.addresses.get`

> * `compute.addresses.delete`

Author: Addressed

* `compute.networks.create`
* `compute.networks.get`
* `compute.networks.list`
* `compute.networks.updatePolicy`
* `compute.routers.create`
* `compute.routers.get`
* `compute.routers.list`
* `compute.routers.update`
* `compute.routes.list`
* `compute.subnetworks.create`
* `compute.subnetworks.get`
* `compute.subnetworks.list`
* `compute.subnetworks.use`
* `compute.subnetworks.useExternalIp`
====

.Required permissions for creating load balancer resources
[%collapsible]
====
* `compute.regionBackendServices.create`
* `compute.regionBackendServices.get`
* `compute.regionBackendServices.list`
* `compute.regionBackendServices.update`
* `compute.regionBackendServices.use`
* `compute.targetPools.addInstance`
* `compute.targetPools.create`
* `compute.targetPools.get`
* `compute.targetPools.list`
* `compute.targetPools.removeInstance`
* `compute.targetPools.use`
====

.Required permissions for creating DNS resources
[%collapsible]
====
* `dns.changes.create`
* `dns.changes.get`
* `dns.managedZones.create`
* `dns.managedZones.get`
* `dns.managedZones.list`
* `dns.networks.bindPrivateDNSZone`
* `dns.resourceRecordSets.create`
* `dns.resourceRecordSets.list`

> ====

Reviewer: ```suggestion
ifdef::template[]
* `dns.resourceRecordSets.update`
endif::template[]
====
```

Author: Addressed

* `compute.machineTypes.list`
====

.Required for creating storage resources
[%collapsible]
====
* `storage.buckets.create`
* `storage.buckets.delete`
* `storage.buckets.get`
* `storage.buckets.list`
* `storage.objects.create`
* `storage.objects.delete`
* `storage.objects.get`
* `storage.objects.list`
====

.Required permissions for creating health check resources
[%collapsible]
====
* `compute.healthChecks.create`
* `compute.healthChecks.get`
* `compute.healthChecks.list`
* `compute.healthChecks.useReadOnly`
* `compute.httpHealthChecks.create`
* `compute.httpHealthChecks.get`
* `compute.httpHealthChecks.list`
* `compute.httpHealthChecks.useReadOnly`
====

.Required permissions to get GCP zone and region related information
[%collapsible]
====
* `compute.globalOperations.get`
* `compute.regionOperations.get`
* `compute.regions.list`
* `compute.zoneOperations.get`
* `compute.zones.get`
* `compute.zones.list`
====

.Required permissions for checking services and quotas
[%collapsible]
====
* `monitoring.timeSeries.list`
* `serviceusage.quotas.get`
* `serviceusage.services.list`
====

.Optional IAM permissions for installation
[%collapsible]
====
* `iam.roles.get`
====

> .Optional images permissions for installation

Reviewer: ```suggestion
.Optional Images permissions for installation
```

Author: Addressed

====

.Optional permission for running gather bootstrap
[%collapsible]
====
* `compute.instances.getSerialPortOutput`
====

.Required permissions for deleting network resources
[%collapsible]
====
* `compute.addresses.delete`
* `compute.addresses.deleteInternal`
* `compute.addresses.list`
* `compute.firewalls.delete`
* `compute.firewalls.list`
* `compute.forwardingRules.delete`
* `compute.forwardingRules.list`
* `compute.networks.delete`
* `compute.networks.list`
* `compute.networks.updatePolicy`
* `compute.routers.delete`
* `compute.routers.list`
* `compute.routes.list`
* `compute.subnetworks.delete`
* `compute.subnetworks.list`
====

.Required permissions for deleting load balancer resources
[%collapsible]
====
* `compute.regionBackendServices.delete`
* `compute.regionBackendServices.list`
* `compute.targetPools.delete`
* `compute.targetPools.list`
====

.Required permissions for deleting DNS resources
[%collapsible]
====
* `dns.changes.create`
* `dns.managedZones.delete`
* `dns.managedZones.get`
* `dns.managedZones.list`
* `dns.resourceRecordSets.delete`
* `dns.resourceRecordSets.list`
ifdef::template[]

> * `dns.resourceRecordSets.update`

Reviewer 2: This permission is required for creating (rather than deleting) DNS resources for user-provisioned GCP infrastructure.

Reviewer: Nice catch. Thanks.

Author: Addressed

* `dns.resourceRecordSets.list`
====

> .Required permissions for deleting service account resources

Reviewer: ```suggestion
.Required permissions for deleting Service Account resources
```

Author: Addressed

[%collapsible]
====
* `iam.serviceAccounts.delete`
* `iam.serviceAccounts.get`
* `iam.serviceAccounts.list`
* `resourcemanager.projects.getIamPolicy`
* `resourcemanager.projects.setIamPolicy`
====

.Required permissions for deleting compute resources
[%collapsible]
====
* `compute.disks.delete`
* `compute.disks.list`
* `compute.instanceGroups.delete`
* `compute.instanceGroups.list`
* `compute.instances.delete`
* `compute.instances.list`
* `compute.instances.stop`
* `compute.machineTypes.list`
====

.Required for deleting storage resources
[%collapsible]
====
* `storage.buckets.delete`
* `storage.buckets.getIamPolicy`
* `storage.buckets.list`
* `storage.objects.delete`
* `storage.objects.list`
====

.Required permissions for deleting health check resources
[%collapsible]
====
* `compute.healthChecks.delete`
* `compute.healthChecks.list`
* `compute.httpHealthChecks.delete`
* `compute.httpHealthChecks.list`
====

> .Optional images permissions for deletion

Reviewer: ```suggestion
.Optional Images permissions for deletion
```

Author: Addressed

* `compute.machineTypes.list`
====

.Required for creating storage resources
[%collapsible]
====
* `storage.buckets.create`
* `storage.buckets.delete`
* `storage.buckets.get`
* `storage.buckets.list`
* `storage.objects.create`
* `storage.objects.delete`
* `storage.objects.get`
* `storage.objects.list`
====

.Required permissions for creating health check resources
[%collapsible]
====
* `compute.healthChecks.create`
* `compute.healthChecks.get`
* `compute.healthChecks.list`
* `compute.healthChecks.useReadOnly`
* `compute.httpHealthChecks.create`
* `compute.httpHealthChecks.get`
* `compute.httpHealthChecks.list`
* `compute.httpHealthChecks.useReadOnly`
====

.Required permissions to get GCP zone and region related information
[%collapsible]
====
* `compute.globalOperations.get`
* `compute.regionOperations.get`
* `compute.regions.list`
* `compute.zoneOperations.get`
* `compute.zones.get`
* `compute.zones.list`
====

.Required permissions for checking services and quotas
[%collapsible]
====
* `monitoring.timeSeries.list`
* `serviceusage.quotas.get`
* `serviceusage.services.list`
====

.Required IAM permissions for installation
[%collapsible]
====
* `iam.roles.get`
====

> ifdef::ipi[]
> .Optional Images permissions for installation
> [%collapsible]
> ====
> * `compute.images.list`
> ====
> endif::ipi[]

Reviewer: ```suggestion
ifdef::ipi[]
.Optional Images permissions for installation
[%collapsible]
====
* `compute.images.list`
====
endif::ipi[]

ifdef::template[]
.Required Images permissions for installation
[%collapsible]
====
* `compute.images.list`
====
endif::template[]
```

Author: Addressed

====

.Required permissions for deleting Service Account resources
[%collapsible]
====
* `iam.serviceAccounts.delete`
* `iam.serviceAccounts.get`
* `iam.serviceAccounts.list`
* `resourcemanager.projects.getIamPolicy`
* `resourcemanager.projects.setIamPolicy`
====

.Required permissions for deleting compute resources
[%collapsible]
====
* `compute.disks.delete`
* `compute.disks.list`
* `compute.instanceGroups.delete`
* `compute.instanceGroups.list`
* `compute.instances.delete`
* `compute.instances.list`
* `compute.instances.stop`
* `compute.machineTypes.list`
====

.Required for deleting storage resources
[%collapsible]
====
* `storage.buckets.delete`
* `storage.buckets.getIamPolicy`
* `storage.buckets.list`
* `storage.objects.delete`
* `storage.objects.list`
====

.Required permissions for deleting health check resources
[%collapsible]
====
* `compute.healthChecks.delete`
* `compute.healthChecks.list`
* `compute.httpHealthChecks.delete`
* `compute.httpHealthChecks.list`
====

ifdef::ipi[]
.Optional Images permissions for deletion
[%collapsible]
====
* `compute.images.list`
====
endif::ipi[]

ifdef::template[]
.Required permissions for deleting image resources
[%collapsible]
====

> * `compute.images.delete`

Author: Addressed

* Service Account User
* Storage Admin

> The following are the required permissions for provisioning GCP infrastructure for creating and deleting the {product-title} cluster.

Author: Addressed

* `compute.networks.get`
* `compute.networks.list`
* `compute.networks.updatePolicy`
* `compute.routers.create`
* `compute.routers.get`
* `compute.routers.list`
* `compute.routers.update`
* `compute.routes.list`
* `compute.subnetworks.create`
* `compute.subnetworks.get`
* `compute.subnetworks.list`
* `compute.subnetworks.use`
* `compute.subnetworks.useExternalIp`
====

.Required permissions for creating load balancer resources
[%collapsible]
====
* `compute.regionBackendServices.create`
* `compute.regionBackendServices.get`
* `compute.regionBackendServices.list`
* `compute.regionBackendServices.update`
* `compute.regionBackendServices.use`
* `compute.targetPools.addInstance`
* `compute.targetPools.create`
* `compute.targetPools.get`
* `compute.targetPools.list`
* `compute.targetPools.removeInstance`
* `compute.targetPools.use`
====

.Required permissions for creating DNS resources
[%collapsible]
====
* `dns.changes.create`
* `dns.changes.get`
* `dns.managedZones.create`
* `dns.managedZones.get`
* `dns.managedZones.list`
* `dns.networks.bindPrivateDNSZone`
* `dns.resourceRecordSets.create`
* `dns.resourceRecordSets.list`
ifdef::template[]
* `dns.resourceRecordSets.update`

> endif::template[]

Author: Addressed

* `compute.machineTypes.list`
====

.Required for creating storage resources
[%collapsible]
====
* `storage.buckets.create`
* `storage.buckets.delete`
* `storage.buckets.get`
* `storage.buckets.list`
* `storage.objects.create`
* `storage.objects.delete`
* `storage.objects.get`
* `storage.objects.list`
====

.Required permissions for creating health check resources
[%collapsible]
====
* `compute.healthChecks.create`
* `compute.healthChecks.get`
* `compute.healthChecks.list`
* `compute.healthChecks.useReadOnly`
* `compute.httpHealthChecks.create`
* `compute.httpHealthChecks.get`
* `compute.httpHealthChecks.list`
* `compute.httpHealthChecks.useReadOnly`
====

.Required permissions to get GCP zone and region related information
[%collapsible]
====
* `compute.globalOperations.get`
* `compute.regionOperations.get`
* `compute.regions.list`
* `compute.zoneOperations.get`
* `compute.zones.get`
* `compute.zones.list`
====

.Required permissions for checking services and quotas
[%collapsible]
====
* `monitoring.timeSeries.list`
* `serviceusage.quotas.get`
* `serviceusage.services.list`
====

.Required IAM permissions for installation
[%collapsible]
====
* `iam.roles.get`
====

> 

Author: Addressed

ifdef::ipi[]
.Optional Images permissions for installation
[%collapsible]
====
* `compute.images.list`
====
endif::ipi[]
ifdef::template[]
.Required Images permissions for installation
[%collapsible]
====
* `compute.images.list`
====

> endif::template[]

Author: Addressed

[%collapsible]
====
* `iam.serviceAccounts.delete`
* `iam.serviceAccounts.get`
* `iam.serviceAccounts.list`
* `resourcemanager.projects.getIamPolicy`
* `resourcemanager.projects.setIamPolicy`
====

.Required permissions for deleting compute resources
[%collapsible]
====
* `compute.disks.delete`
* `compute.disks.list`
* `compute.instanceGroups.delete`
* `compute.instanceGroups.list`
* `compute.instances.delete`
* `compute.instances.list`
* `compute.instances.stop`
* `compute.machineTypes.list`
====

.Required for deleting storage resources
[%collapsible]
====
* `storage.buckets.delete`
* `storage.buckets.getIamPolicy`
* `storage.buckets.list`
* `storage.objects.delete`
* `storage.objects.list`
====

.Required permissions for deleting health check resources
[%collapsible]
====
* `compute.healthChecks.delete`
* `compute.healthChecks.list`
* `compute.httpHealthChecks.delete`
* `compute.httpHealthChecks.list`
====

ifdef::ipi[]
.Optional Images permissions for deletion
[%collapsible]
====
* `compute.images.list`
====
endif::ipi[]

ifdef::template[]
.Required permissions for deleting image resources
[%collapsible]
====
* `compute.images.delete`
* `compute.images.list`
====

> endif::template[]

Author: Addressed

[%collapsible]
====
* `dns.changes.create`
* `dns.managedZones.delete`
* `dns.managedZones.get`
* `dns.managedZones.list`
* `dns.resourceRecordSets.delete`
* `dns.resourceRecordSets.list`
====

.Required permissions for deleting Service Account resources
[%collapsible]
====
* `iam.serviceAccounts.delete`
* `iam.serviceAccounts.get`
* `iam.serviceAccounts.list`
* `resourcemanager.projects.getIamPolicy`
* `resourcemanager.projects.setIamPolicy`
====

.Required permissions for deleting compute resources
[%collapsible]
====
* `compute.disks.delete`
* `compute.disks.list`
* `compute.instanceGroups.delete`
* `compute.instanceGroups.list`
* `compute.instances.delete`
* `compute.instances.list`
* `compute.instances.stop`
* `compute.machineTypes.list`
====

.Required for deleting storage resources
[%collapsible]
====
* `storage.buckets.delete`
* `storage.buckets.getIamPolicy`
* `storage.buckets.list`
* `storage.objects.delete`
* `storage.objects.list`
====

.Required permissions for deleting health check resources
[%collapsible]
====
* `compute.healthChecks.delete`
* `compute.healthChecks.list`
* `compute.httpHealthChecks.delete`
* `compute.httpHealthChecks.list`
====

> endif::ipiupi[]
> 
> ifdef::ipi[]
> .Optional Images permissions for deletion
> [%collapsible]
> ====
> * `compute.images.list`
> ====
> endif::ipi[]
> 
> ifdef::upi[]
> .Required permissions for deleting image resources
> [%collapsible]
> ====
> * `compute.images.delete`
> * `compute.images.list`
> ====
> endif::upi[]

Reviewer: >  With this said, suggest to rename ``Optional Images permissions for deletion`` into ``Required Images permissions for deletion``. WDYT? Thanks!

@xenolinux Following changes will be required to incorporate the above.

```suggestion

.Required permissions for deleting image resources
[%collapsible]
====
* `compute.images.delete`
* `compute.images.list`
====
endif::ipiupi[]
```

Reviewer 2: Sorry, don't quite understand. IPI only requires `compute.images.list` permission during uninstallation, the other permission `compute.images.delete` is not needed during IPI installation or uninstallation. Could you clarify a bit? Thanks in advance!

Reviewer: Ok my bad. I will update the suggested changes. Thanks for pointing that out @jianli-wei .

Reviewer: Updated the suggestion. @jianli-wei Let me know if the suggestion looks fine.

Author: Addressed

Reviewer 2: Yes, it looks fine now. 

But, as for the `Required Images permissions` and `Required Images permissions for installation` in UPI part, I suggest to move the permissions of the former to the latter, and remove the former section, because those 4 permissions are for installation. Thanks!

Author: @jianli-wei Addressed

Reviewer 2: Sorry, please help removing one duplicated `compute.images.list` from the list, thanks!

Author: @jianli-wei Done

Reviewer 2: Please also sort those 4 permissions, thanks!

Reviewer 2: @xenolinux

Author: @jianli-wei I believe its - `create`, `delete`, `get`, `list` - sorted in that manner

Reviewer 2: Correct, thanks!

.Required roles for user-provisioned GCP infrastructure
* Deployment Manager Editor
* Service Account Key Admin

> The following are the required permissions for user-provisioned GCP infrastructure for creating and deleting the {product-title} cluster.

Reviewer: ```suggestion
The following are the additional permissions required for user-provisioned GCP infrastructure for creating and deleting the {product-title} cluster.
```

Author: Acked

* `compute.regionBackendServices.list`
* `compute.regionBackendServices.update`
* `compute.regionBackendServices.use`
* `compute.targetPools.addInstance`
* `compute.targetPools.create`
* `compute.targetPools.get`
* `compute.targetPools.list`
* `compute.targetPools.removeInstance`
* `compute.targetPools.use`
====

.Required permissions for creating DNS resources
[%collapsible]
====
* `dns.changes.get`
* `dns.changes.create`
* `dns.managedZones.create`
* `dns.managedZones.delete`
* `dns.managedZones.get`
* `dns.managedZones.list`
* `dns.networks.bindPrivateDNSZone`
* `dns.resourceRecordSets.create`
* `dns.resourceRecordSets.delete`
* `dns.resourceRecordSets.list`
* `dns.resourceRecordSets.update`
====

.Required permissions for creating Service Account resources
[%collapsible]
====
* `iam.serviceAccountKeys.create`
* `iam.serviceAccountKeys.delete`
* `iam.serviceAccountKeys.get`
* `iam.serviceAccountKeys.list`
* `iam.serviceAccounts.actAs`
* `iam.serviceAccounts.create`
* `iam.serviceAccounts.get`
* `iam.serviceAccounts.list`
* `resourcemanager.projects.get`
* `resourcemanager.projects.getIamPolicy`
====

.Required permissions for creating compute resources

> [%collapsible]
> ====
> * `compute.disks.create`
> * `compute.disks.get`
> * `compute.disks.list`
> * `compute.instanceGroups.create`
> * `compute.instanceGroups.get`
> * `compute.instanceGroups.list`
> * `compute.instanceGroups.update`
> * `compute.instanceGroups.use`
> * `compute.instances.create`
> * `compute.instances.get`
> * `compute.instances.list`
> * `compute.instances.setLabels`
> * `compute.instances.setMetadata`
> * `compute.instances.setServiceAccount`
> * `compute.instances.setTags`
> * `compute.instances.use`
> * `compute.machineTypes.get`
> * `compute.machineTypes.list`
> ====
> 
> .Required for creating storage resources
> [%collapsible]
> ====
> * `storage.buckets.create`
> * `storage.buckets.get`
> * `storage.buckets.list`
> * `storage.objects.create`
> * `storage.objects.delete`
> * `storage.objects.get`
> * `storage.objects.list`
> ====
> 
> .Required permissions for creating health check resources
> [%collapsible]
> ====
> * `compute.healthChecks.create`
> * `compute.healthChecks.get`
> * `compute.healthChecks.list`
> * `compute.healthChecks.useReadOnly`
> * `compute.httpHealthChecks.create`
> * `compute.httpHealthChecks.get`
> * `compute.httpHealthChecks.list`
> * `compute.httpHealthChecks.useReadOnly`
> ====
> 
> .Additional permissions for checking services and quotas
> [%collapsible]
> ====
> * `monitoring.timeSeries.list`
> * `serviceusage.quotas.get`
> * `serviceusage.services.list`
> ====
> 
> .Additional IAM permissions for installation
> [%collapsible]
> ====
> * `iam.roles.get`
> ====

Reviewer: ```suggestion

```

Author: Acked

* `iam.serviceAccountKeys.get`
* `iam.serviceAccountKeys.list`
* `iam.serviceAccounts.actAs`
* `iam.serviceAccounts.create`
* `iam.serviceAccounts.delete`
* `iam.serviceAccounts.get`
* `iam.serviceAccounts.list`
* `resourcemanager.projects.get`
* `resourcemanager.projects.getIamPolicy`
* `resourcemanager.projects.setIamPolicy`
====

.Required permissions for deleting network resources
[%collapsible]
====
* `compute.addresses.delete`
* `compute.addresses.deleteInternal`
* `compute.addresses.list`
* `compute.firewalls.delete`
* `compute.firewalls.list`
* `compute.forwardingRules.delete`
* `compute.forwardingRules.list`
* `compute.networks.delete`
* `compute.networks.list`
* `compute.networks.updatePolicy`
* `compute.routers.delete`
* `compute.routers.list`
* `compute.routes.list`
* `compute.subnetworks.delete`
* `compute.subnetworks.list`
====

.Required permissions for deleting load balancer resources
[%collapsible]
====
* `compute.regionBackendServices.delete`
* `compute.regionBackendServices.list`
* `compute.targetPools.delete`
* `compute.targetPools.list`
====

.Required permissions for deleting DNS resources
[%collapsible]
====
* `dns.changes.create`

> * `dns.managedZones.delete`
> * `dns.managedZones.get`
> * `dns.managedZones.list`
> * `dns.resourceRecordSets.delete`
> * `dns.resourceRecordSets.list`
> ====
> 
> .Required permissions for deleting Service Account resources
> [%collapsible]
> ====
> * `iam.serviceAccounts.delete`
> * `iam.serviceAccounts.get`
> * `iam.serviceAccounts.list`
> * `resourcemanager.projects.getIamPolicy`
> * `resourcemanager.projects.setIamPolicy`
> ====
> 
> .Required permissions for deleting compute resources
> [%collapsible]
> ====
> * `compute.disks.delete`
> * `compute.disks.list`
> * `compute.instanceGroups.delete`
> * `compute.instanceGroups.list`
> * `compute.instances.delete`
> * `compute.instances.list`
> * `compute.instances.stop`
> * `compute.machineTypes.list`
> ====
> 
> .Required for deleting storage resources
> [%collapsible]
> ====
> * `storage.buckets.delete`
> * `storage.buckets.getIamPolicy`
> * `storage.buckets.list`
> * `storage.objects.delete`
> * `storage.objects.list`
> ====
> 
> .Required permissions for deleting health check resources
> [%collapsible]
> ====
> * `compute.healthChecks.delete`
> * `compute.healthChecks.list`
> * `compute.httpHealthChecks.delete`
> * `compute.httpHealthChecks.list`
> ====
> 
> .Additional Images permissions for installation
> [%collapsible]
> ====
> * `compute.images.list`
> ====
> 
> .Required permissions for deleting image resources
> [%collapsible]
> ====
> * `compute.images.delete`
> ====

Reviewer: ```suggestion

```

Author: Acked

.Required Images permissions
[%collapsible]
====
* `compute.images.create`
* `compute.images.delete`
* `compute.images.get`
* `compute.images.list`
====

.Required permissions to get Region related information
[%collapsible]
====
* `compute.regions.get`
====

.Required Deployment Manager permissions
[%collapsible]
====
* `deploymentmanager.deployments.create`
* `deploymentmanager.deployments.delete`
* `deploymentmanager.deployments.get`
* `deploymentmanager.deployments.list`
* `deploymentmanager.manifests.get`
* `deploymentmanager.operations.get`
* `deploymentmanager.resources.list`
====

.Required Service Account permissions
[%collapsible]
====
* `iam.serviceAccountKeys.create`
* `iam.serviceAccountKeys.delete`
* `iam.serviceAccountKeys.get`
* `iam.serviceAccountKeys.list`
* `iam.serviceAccounts.actAs`
* `iam.serviceAccounts.create`
* `iam.serviceAccounts.delete`
* `iam.serviceAccounts.get`
* `iam.serviceAccounts.list`
* `resourcemanager.projects.get`
* `resourcemanager.projects.getIamPolicy`
* `resourcemanager.projects.setIamPolicy`
====

.Required permissions for deleting Deployment Manager
[%collapsible]
====
* `deploymentmanager.deployments.delete`

> * `deploymentmanager.deployments.list`
> * `deploymentmanager.operations.get`

Reviewer: ```suggestion
* `deploymentmanager.operations.get`
* `deploymentmanager.deployments.list`
```
Rearranging the permissions in ascending order

Author: Addressed

.Required roles for user-provisioned GCP infrastructure
* Deployment Manager Editor
* Service Account Key Admin

The following are the additional permissions required for user-provisioned GCP infrastructure for creating and deleting the {product-title} cluster.

.Required Images permissions
[%collapsible]
====
* `compute.images.create`
* `compute.images.delete`
* `compute.images.get`
* `compute.images.list`
====

.Required permissions to get Region related information
[%collapsible]
====
* `compute.regions.get`
====

> .Required deployment manager permissions

Reviewer: @xenolinux Any specific reason you changed the  ``Deployment Manager`` to ``deployment manager``?

Reviewer: Deployment Manager is a GCP specific term. The existing documentation of UPI installation for GCP has used the term ``Deployment Manager`` and not ``deployment manager``. https://docs.openshift.com/container-platform/4.12/installing/installing_gcp/installing-gcp-user-infra.html

Author: I referred to https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#general-references and https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc and thought it should be in small letters. But yes you are right. If that's the case then we should keep it in caps

Author: Reverted the changes - Addressed

.Required roles for user-provisioned GCP infrastructure
* Deployment Manager Editor
* Service Account Key Admin

The following are the additional permissions required for user-provisioned GCP infrastructure for creating and deleting the {product-title} cluster.

.Required Images permissions
[%collapsible]
====
* `compute.images.create`
* `compute.images.delete`
* `compute.images.get`
* `compute.images.list`
====

.Required permissions to get Region related information
[%collapsible]
====
* `compute.regions.get`
====

.Required Deployment Manager permissions
[%collapsible]
====
* `deploymentmanager.deployments.create`
* `deploymentmanager.deployments.delete`
* `deploymentmanager.deployments.get`
* `deploymentmanager.deployments.list`
* `deploymentmanager.manifests.get`
* `deploymentmanager.operations.get`
* `deploymentmanager.resources.list`
====

> .Required Service Account permissions

Reviewer 2: IPI part has the section already, so please remove the section for user-provisioned GCP infrastructure.

Author: Addressed- removed

[%collapsible]
====
* `iam.serviceAccountKeys.create`
* `iam.serviceAccountKeys.delete`
* `iam.serviceAccountKeys.get`
* `iam.serviceAccountKeys.list`
* `iam.serviceAccounts.actAs`
* `iam.serviceAccounts.create`
* `iam.serviceAccounts.delete`
* `iam.serviceAccounts.get`
* `iam.serviceAccounts.list`
* `resourcemanager.projects.get`
* `resourcemanager.projects.getIamPolicy`
* `resourcemanager.projects.setIamPolicy`
====

> .Required permissions for deleting Deployment Manager

Reviewer: This section documents permissions for both creating and deleting resources. I would suggest to remove ``Required permissions for deleting Deployment Manager`` from here.

> .Required permissions for deleting Deployment Manager

Author: Addressed

[%collapsible]
====
* `deploymentmanager.deployments.delete`
* `deploymentmanager.operations.get`

> * `deploymentmanager.deployments.list`

Author: Addressed

.Required roles for user-provisioned GCP infrastructure
* Deployment Manager Editor
* Service Account Key Admin

> The following are the additional permissions required for user-provisioned GCP infrastructure for creating and deleting the {product-title} cluster.

Reviewer: ```suggestion
ifdef::upi[]
The following are the additional permissions required for user-provisioned GCP infrastructure for creating and deleting the {product-title} cluster.
```

Author: Addressed

.Required Images permissions
[%collapsible]
====
* `compute.images.create`
* `compute.images.delete`
* `compute.images.get`
* `compute.images.list`
====

.Required permissions to get Region related information
[%collapsible]
====
* `compute.regions.get`
====

.Required Deployment Manager permissions
[%collapsible]
====
* `deploymentmanager.deployments.create`
* `deploymentmanager.deployments.delete`
* `deploymentmanager.deployments.get`
* `deploymentmanager.deployments.list`
* `deploymentmanager.manifests.get`
* `deploymentmanager.operations.get`
* `deploymentmanager.resources.list`

> ====

Reviewer: ```suggestion
====
endif::upi[]
```

Author: Addressed

ifeval::["{context}" == "installing-gcp-user-infra-vpc"]
:!template:
endif::[]
ifeval::["{context}" == "installing-gcp-account"]
:!ipi:

> endif::[]

Reviewer: ```suggestion
endif::[]
ifeval::["{context}" == "installing-gcp-user-infra"]
:!upi:
endif::[]
ifeval::["{context}" == "installing-gcp-account"]
:!ipiupi:
endif::[]
ifeval::["{context}" == "installing-gcp-user-infra"]
:!ipiupi:
endif::[]
```

Author: Addressed
